packages/service/ni_measurement_plugin_sdk_service/grpc/channelpool.py

```diff
@@ -78,3 +73,43 @@ def close(self) -> None:
             for channel in self._channel_cache.values():
                 channel.close()
             self._channel_cache.clear()
+
+    def _create_channel(self, target: str) -> grpc.Channel:
+        options = [
+            ("grpc.max_receive_message_length", -1),
+            ("grpc.max_send_message_length", -1),
+        ]
+        if self._is_local(target):
+            options.append(("grpc.enable_http_proxy", 0))
+        channel = grpc.insecure_channel(target, options)
+        if ClientLogger.is_enabled():
+            channel = grpc.intercept_channel(channel, ClientLogger())
+        return channel
+
+    def _is_local(self, target: str) -> bool:
+        hostname = ""
+        # First, check if the target string is in URL format
+        parse_result = urlparse(target)
+        if parse_result.scheme and parse_result.hostname and parse_result.port:
+            hostname = parse_result.hostname
+        else:
+            # Next, check for target string in <host_name>:<port> format
+            match = re.match(r"^(.*):(\d+)$", target)
+            if match:
+                hostname = match.group(1)
+
+        if not hostname:
+            return False
+        if hostname == "localhost" or hostname == "LOCALHOST":
+            return True
+
+        # IPv6 addresses don't support parsing with leading/trailing brackets so we need to remove them.
```

Comment:
```
./ni_measurement_plugin_sdk_service/grpc/channelpool.py:106:101: W505 doc line too long (104 > 100 characters)
```
Run `poetry run ni-python-styleguide fix`. If it doesn't update this, you may need to wrap it manually.

Reply:
Manually fixed since `ni-python-styleguide fix` didn't correct it. Is there something I should have run that would have detected that? Or maybe I just need to configure some editor settings?

Reply:
`poetry run ni-python-styleguide lint`

I need to update https://github.com/ni/measurement-plugin-python/blob/main/CONTRIBUTING.md .